main.go

```diff
@@ -456,6 +467,79 @@ func (a *arg) list(proxy string) error {
 			return nil
 		}
 
+		seen := seenMods[pkg.Module.Path]
+		if !seen && len(seenMods) > 0 {
+			a.resErr = errMultiModules
+			return nil
+		}
+
+		seenMods[pkg.Module.Path] = true
+
+		// If we are not in main-module mode (i.e. -m is not provided), then we
+		// are working in a temporary module. Any replacements in
+		// $(mp.Module.GoMod) need to be applied to $(pkg.wd)/go.mod. So the
+		// simplest thing to do is copy the main package's module's go.mod over
+		// the top of the temporary module's go.mod and then adjust the module
+		// line and add the single requirement that we now have resolved.
+		if !seen && !*fMainMod {
+			srcPath := pkg.Module.GoMod
+			src, err := os.Open(srcPath)
+			if err != nil {
+				return fmt.Errorf("failed to open src %v: %v", srcPath, err)
+			}
+
+			destPath := filepath.Join(a.wd, "go.mod")
+			dest, err := os.Create(destPath)
+			if err != nil {
+				src.Close()
```

Comment:
I think you could delete this line - I understand that you don't want to use defers since this is a loop, but you stop at any error, in which case it doesn't really matter if we left one file open. The entire program will exit.

Reply:
Yep, good point. Thanks